crates/rpc-types/src/eth/block.rs

```diff
@@ -75,10 +75,13 @@ pub struct Header {
     /// Difficulty
     pub difficulty: U256,
     /// Block number
+    #[serde(default, with = "alloy_serde::num::u64_hex_opt")]
     pub number: Option<u64>,
     /// Gas Limit
+    #[serde(default, with = "alloy_serde::num::u128_hex_or_decimal")]
     pub gas_limit: u128,
     /// Gas Used
+    #[serde(default, with = "alloy_serde::num::u128_hex_or_decimal")]
```

Comment:
this should just be hex

Reply:
Going to address this in a separate PR for https://github.com/alloy-rs/alloy/issues/465